File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> v(128,0);
        int i=0,j=0,ans=0;
        while(i<s.size() && j<s.size())
        {
            if(v[s[j]]==0)
            {
                v[s[j]]++;
                j++;
                ans=max(ans,j-i);
            }
            else{
                v[s[j]]++;
                while(v[s[j]]>1)
                {
                    v[s[i]]--;i++;
                }j++;
            }
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] map = new int[128];
        int n=s.length();
        if(n==0)
        {
            return 0;
        }
        int ans=1;
        int i=0,j=1;
        Arrays.fill(map,0);
        map[s.charAt(i)]++;
        while(j<n)
        {
            char c=s.charAt(j);
            while(map[c]!=0)
            {
                map[s.charAt(i)]--;i++;
            }
            map[c]++;
            ans=Math.max(ans,j-i+1);
            j++;
        }

        return ans;
    }
}